### PR TITLE
feat: add delete on op-msg

### PR DESCRIPTION
**Pull request by NikolayBaranovv**

### CODE REVIEW

File: txmongo/collection.py [hunk 1]
```diff
@@ -1238,49 +1238,67 @@ def on_proto(proto):
...
+        let: Optional[dict] = None,
+        _deadline: Optional[float] = None,
+    ) -> Deferred[DeleteResult]:
         """delete_one(filter)"""
-
-        def on_ok(raw_response):
-            return DeleteResult(raw_response, self.write_concern.acknowledged)
-
-        return self._delete(filter, False, _deadline).addCallback(on_ok)
+        return self._delete(filter, let=let, multi=False, _deadline=_deadline)
```
Comment (IlyaSkriblovsky) [inline txmongo/collection.py]:
```suggestion
        return self._delete(filter, let, multi=False, _deadline=_deadline)
```

File: txmongo/collection.py [hunk 2]
```diff
@@ -1238,49 +1238,67 @@ def on_proto(proto):
...
+        let: Optional[dict] = None,
+        _deadline: Optional[float] = None,
+    ) -> Deferred[DeleteResult]:
         """delete_many(filter)"""
-
-        def on_ok(raw_response):
-            return DeleteResult(raw_response, self.write_concern.acknowledged)
-
-        return self._delete(filter, True, _deadline).addCallback(on_ok)
+        return self._delete(filter, let=let, multi=True, _deadline=_deadline)
```
Comment (IlyaSkriblovsky) [inline txmongo/collection.py]:
```suggestion
        return self._delete(filter, let, multi=True, _deadline=_deadline)
```

File: txmongo/protocol.py [hunk 3]
```diff
@@ -404,6 +404,10 @@ def opcode(self):
         #        Maybe we should refactor other messages too. Or get rid of MsgHeader dataclass.
         return self.header.opcode
 
+    @classmethod
+    def create_flag_bits(cls, not_more_to_come: bool ) -> int:
```
Comment (IlyaSkriblovsky) [inline txmongo/protocol.py]: How about `black` in `pre-commit`? :)

```suggestion
    def create_flag_bits(cls, not_more_to_come: bool) -> int:
```

File: txmongo/collection.py [hunk 4]
```diff
@@ -1238,49 +1238,67 @@ def on_proto(proto):
 
         return self._database.connection.getprotocol().addCallback(on_proto)
 
-    def _delete(self, filter, multi, _deadline):
+    @defer.inlineCallbacks
+    def _delete(
+        self, filter: dict, let: Optional[dict], multi: bool, _deadline: Optional[float]
```
Comment (IlyaSkriblovsky) [inline txmongo/collection.py:1041]: Please add tests for newly added `let` argument. Or at least add it to the plan in `notes.md`.

**Review by IlyaSkriblovsky — CHANGES_REQUESTED** [review]

File: txmongo/protocol.py (hunk 3 above)
Comment (NikolayBaranovv) [reply txmongo/protocol.py]: need install pre-commit on every machine!)

File: txmongo/collection.py (hunk 4 above)
Comment (NikolayBaranovv) [reply txmongo/collection.py:1041]: test added in TestDeleteMany

**Pull request merged**